go/interpreter/lfvm/memory.go

```diff
@@ -121,16 +121,14 @@ func (m *Memory) SetByte(offset uint64, value byte, c *context) error {
 	return nil
 }
 
-func (m *Memory) SetWord(offset uint64, value *uint256.Int, c *context) error {
+// setWord sets a 32-byte word at the given offset, expanding memory as needed and charging for it.
+// Returns error if insufficient gas or offset+32 overflows.
+func (m *Memory) setWord(offset uint64, value *uint256.Int, c *context) error {
```

Comment:
Once #759 is merged, the body of this function could likely be simplified to something like
```
data := value.Bytes32()
return m.set(offset, 32, data[:], c)
```
Whether we should go for this depends on whether the promised speed-up statement below still holds up. Would be interesting to add a micro-benchmark comparing the 2-line version listed above with the unrolled version that is implemented here right now.

Reply:
I experimented with the following benchmark:
```
func BenchmarkMemory_SetWord(b *testing.B) {
	m := NewMemory()
	m.expandMemoryWithoutCharging(32)
	ctxt := getEmptyContext()
	word := new(uint256.Int)
	for i := 0; i < b.N; i++ {
		m.SetWord(0, word, &ctxt)
	}
}
```
and the results for the current version vs. the 2-line version looks like this:
```
goos: linux
goarch: amd64
pkg: github.com/Fantom-foundation/Tosca/go/interpreter/lfvm
cpu: Intel(R) Core(TM) i7-5820K CPU @ 3.30GHz
                  │ specialized.txt │            generic.txt             │
                  │     sec/op      │   sec/op     vs base               │
Memory_SetWord-12       15.56n ± 4%   16.99n ± 2%  +9.19% (p=0.000 n=10)
```
The difference is ~1.5 nano seconds, far from the claimed 7x improvement. I would argue that this small difference (in absolute terms) is not worth the extra complexity. Wdyt?

Reply:
I did not change it originally because of the promised speedup. I should have made the benchmark an tested it, yes. I would say it depends on how often is mload8 called. do we have this statistic ?

Reply:
the number of times this `MSTORE8` is called is not significant enough for this performance degradation to have any impact. I will do as you suggested and implment `setWord` as a specialization of `set`, just as `setByte` is in #759